creator/lib/screens/login_screen.dart

```diff
@@ -8,10 +17,221 @@ class LoginScreen extends StatefulWidget {
 }
 
 class _LoginScreenState extends State<LoginScreen> {
+  bool _loginInProgress = false;
+
+  // ログインボタン
+  Widget _buildLoginButtons(BuildContext context) {
+    return Container(
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          _buildLoginButton(
+            context,
+            'assets/signin_button/google_logo.png',
+            'Googleでログイン',
+            () {
+              _tryLoginWith(
+                context,
+                signInWithGoogle(),
+              );
+            },
+          ),
+          SizedBox(
+            height: 16,
+          ),
+          _buildLoginButton(
+            context,
+            'assets/signin_button/apple_logo.png',
+            'Appleでログイン',
+            () {
+              _tryLoginWith(
+                context,
+                signInWithApple(),
+              );
+            },
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildLoginButton(
+    BuildContext context,
+    String asset,
+    String text,
+    Function() method,
+  ) {
+    final mq = MediaQuery.of(context).size;
+    final screenWidth = mq.width;
+    final screenHeight = mq.height;
+
+    return Center(
+      child: Container(
+        width: screenWidth < 600 ? screenWidth * 0.8 : screenWidth * 0.5,
+        height: screenHeight * 0.06,
+        child: ElevatedButton(
+          style: Theme.of(context).elevatedButtonTheme.style?.copyWith(
+                backgroundColor: MaterialStateProperty.all(Colors.white),
+                foregroundColor: MaterialStateProperty.all(Colors.black),
+              ),
+          onPressed: method,
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              _buildLoginMethodLogo(asset),
+              SizedBox(
+                width: 16,
+              ),
+              Text(
+                text,
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  Container _buildLoginMethodLogo(String asset) {
+    final iconsSize = MediaQuery.of(context).size.width * 0.08;
+    return Container(
+      width: iconsSize,
+      height: iconsSize,
+      child: FittedBox(
+        fit: BoxFit.cover,
+        child: Image.asset(asset),
+      ),
+    );
+  }
+
+  // ログインできていたらHome、　出来ていなかったらsalon_applying_screen
+
+  //TODO: _addToDatabaseでデータベースに登録するが、現時点では実装していない。
+  void _navigateBasedOnRole() {
+    final _auth = FirebaseAuth.instance.currentUser;
+    print('宏輝: $_auth');
+    FirebaseFirestore.instance
+        .collection('users')
+        .doc(_auth?.email)
+        .get()
+        .then((DocumentSnapshot snapshot) {
```

Comment:
Please use DbHandler to get UserModel and Salon objects and use them instead

Reply:
I add new function to database.dart and I used dbHandler.